In [1]:
# importing necessary libraries

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

In [6]:
# loading the downloaded csv file
df = pd.read_csv('who_suicide_statistics.csv')


In [7]:
# first five rows
df.head()

country  year     sex          age  suicides_no  population
0  Albania  1985  female  15-24 years          NaN    277900.0
1  Albania  1985  female  25-34 years          NaN    246800.0
2  Albania  1985  female  35-54 years          NaN    267500.0
3  Albania  1985  female   5-14 years          NaN    298300.0
4  Albania  1985  female  55-74 years          NaN    138700.0

In [32]:
# checking the column names

print("The name of the column is: ")
for column in df.columns:
    print(f"'{column}' ")



The name of the column is: 
'country' 
'year' 
'sex' 
'age' 
'suicides_no' 
'population' 


In [44]:
# checking the data type of each column

print("The data type of each column is: ")
df.dtypes

The data type of each column is: 


country         object
year             int64
sex             object
age             object
suicides_no    float64
population     float64
dtype: object

In [53]:
# check for the number of rows and columns

# The number of rows in dataset is 
print(f"The number of rows in our dataset is =  '{df.shape[0]}' \n")
# The number of column in our dataset is
print(f"The number of columns in our dataset is = '{ df.shape[1]}'")

The number of rows in our dataset is =  '43776' 

The number of columns in our dataset is = '6'


---
### **Observation:** 

- Our dataset has 6 coulmns in which three of them are catagorical and three are numeric.

- Columns like `country` , 'sex' , 'age' have object data type whereas `year` has int data type. In contrast to that both `suicides_no` and `population` have float data type.

- The number of rows in our dataset is 43776
---

In [58]:
# check for the value counts in country column

df['country'].nunique()

141

---

We have a record of total 141 countries across the globe.

---

In [56]:
# check for the unique values in each column

df['year'].unique()

array([1985, 1986, 1987, 1988, 1989, 1990, 1991, 1992, 1993, 1994, 1995,
       1996, 1997, 1998, 1999, 2000, 2001, 2002, 2003, 2004, 2005, 2006,
       2007, 2008, 2009, 2010, 2011, 2012, 2013, 2014, 2015, 1983, 1984,
       2016, 1979, 1980, 1981, 1982], dtype=int64)

In [66]:
# getting count of unqiue values in year column

df.year.nunique()

38

---

We have data which spans over 38 years since 1985. to 2016


---

In [59]:
# check for the unique values count in sex column

df.sex.value_counts()

sex
female    21888
male      21888
Name: count, dtype: int64

---



In [60]:
# check for how many unique age groups we have

print(f'{df['age'].unique()}')
# df.age.unique()

array(['15-24 years', '25-34 years', '35-54 years', '5-14 years',
       '55-74 years', '75+ years'], dtype=object)

In [61]:
# checking the count of unique values of each age group

df.age.value_counts()

age
15-24 years    7296
25-34 years    7296
35-54 years    7296
5-14 years     7296
55-74 years    7296
75+ years      7296
Name: count, dtype: int64

In [63]:
# check for the unique values in suicides_no column 

df.suicides_no.unique()

array([       nan, 1.4000e+01, 4.0000e+00, ..., 1.1634e+04, 9.0680e+03,
       3.1710e+03])

In [64]:
# check the sum of null values in suicides_no column

df.suicides_no.isnull().sum()     

2256

In [65]:
# applying descriptive statistics

df.describe()

year   suicides_no    population
count  43776.000000  41520.000000  3.831600e+04
mean    1998.502467    193.315390  1.664091e+06
std       10.338711    800.589926  3.647231e+06
min     1979.000000      0.000000  2.590000e+02
25%     1990.000000      1.000000  8.511275e+04
50%     1999.000000     14.000000  3.806550e+05
75%     2007.000000     91.000000  1.305698e+06
max     2016.000000  22338.000000  4.380521e+07